prio/server.c

```diff
@@ -83,6 +83,19 @@ PrioServer_aggregate(PrioServer s, PrioVerifier v)
   return MPArray_addmod(s->data_shares, arr, &s->cfg->modulus);
 }
 
+SECStatus
+PrioServer_merge(PrioServer s, const_PrioServer s_i)
+{
+  if (s->cfg->num_data_fields != s_i->cfg->num_data_fields ||
+      mp_cmp(&s->cfg->modulus, &s_i->cfg->modulus) ||
+      strncmp((const char*)s->cfg->batch_id,
+              (const char*)s_i->cfg->batch_id,
+              s->cfg->batch_id_len)) {
```

Comment:
I think you want to check that `s->cfg` and `s_i->cfg` have the same `batch_id_len` here first. Otherwise, the string comparison can run off of the end of one of the two `batch_id` strings.